src/coreclr/src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -9215,32 +9273,80 @@ HRESULT ProfToEEInterfaceImpl::GetRuntimeInformation(USHORT * pClrInstanceId,
         return E_INVALIDARG;
     }
 
-    if (pcchVersionString != NULL)
+    // Try and get the version set by the host
+    LPCWSTR fxProductVersion = Configuration::GetKnobStringValue(W("FX_PRODUCT_VERSION"));
+    ULONG cchFxProductVersion =  fxProductVersion == NULL ? 0 : (ULONG)wcslen(fxProductVersion) + 1;
+
+    if (pcchVersionString != NULL || szVersionString != NULL)
     {
-        HRESULT hr = GetCORVersionInternal(szVersionString, (DWORD)cchVersionString, (DWORD *)pcchVersionString);
-        if (FAILED(hr))
-            return hr;
+        if (fxProductVersion != NULL)
+        {
+            if (pcchVersionString != NULL)
+            {
+                *pcchVersionString = cchFxProductVersion;
+            }
+
+            if (cchFxProductVersion >= cchVersionString)
+            {
+                return HRESULT_FROM_WIN32(ERROR_INSUFFICIENT_BUFFER);
+            }
+
+            if (szVersionString != NULL)
+            {
+                CopyMemory(szVersionString, fxProductVersion, cchFxProductVersion * sizeof(WCHAR));
+            }
+        }
+        else
+        {
+            // The host hasn't set a version, fall back to the compile time ones
+            HRESULT hr = GetCORVersionInternal(szVersionString, (DWORD)cchVersionString, (DWORD *)pcchVersionString);
+            if (FAILED(hr))
+            {
+                return hr;
+            }
+        }
+    }
+
+    // defaults to use if the host hasn't set a version to parse
+    USHORT majorVersion = CLR_MAJOR_VERSION;
```

Comment:
I think falling back to this version may cause more confusion than benefit. The profiler probably isn't in a position to distinguish which is which and interpreting one as the other is likely to produce bad versioning decisions.
What if we defined the function returns FX_PRODUCT_VERSION when present and a NULL string/0 version when it isn't?